modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/idempotency/NuxeoIdempotentResponse.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.ecm.platform.web.common.idempotency;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletResponse;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+/**
+ * POJO representing response to be serialized and served by {@link NuxeoIdempotentFilter}.
+ *
+ * @since 11.5
+ */
+public class NuxeoIdempotentResponse {
+
+    protected int status;
+
+    protected Map<String, Collection<String>> headers = new LinkedHashMap<>();
+
+    public int getStatus() {
+        return status;
+    }
+
+    public void setStatus(int status) {
+        this.status = status;
+    }
+
+    public Map<String, Collection<String>> getHeaders() {
+        return headers;
+    }
+
+    public void setHeader(String name, Collection<String> value) {
+        headers.put(name, value);
+    }
+
+    protected static final ObjectMapper getMapper() {
```

Comment:
You should put the mapper in a static final field instead of constructing it every time.